### PR TITLE
Confirm email addresses

### LINKED ISSUE #184: Email Validation
Would be helpful to TD's to send an email validation link so we can ensure that people input their email correctly and that they own them

### PR DESCRIPTION
**Pull request by jaredbeck**
Closes #184

I've done some very limited testing in local dev. I'd recommend further manual testing of the following:

- sign up -> confirm -> sign in
- sign in without confirming, should get message like "sorry, you're inactive"
- change email -> confirm -> sign in with new email
- change email -> don't confirm -> sign in with old email
- change email -> receive warning that it changed delivered to old address
- change password

The default wording is used for the various emails and flash messages. This wording can be customized in future PRs.

### CODE REVIEW

File: spec/requests/api_spec.rb [hunk 1]
```diff
@@ -55,6 +55,11 @@
 
     describe "GET KGS Username" do
       it "returns a 200 status code for a username that does exist" do
+        skip <<-EOS.squish.freeze
+          Intermittently failing with "OpenURI::HTTPError: 503". We should
+          rewrite this test to avoid actual network operations, by using
+          rspec-mocks, or more advanced techniques like webmock or vcr.
+        EOS
```
Comment (neagle) [inline spec/requests/api_spec.rb:62]: Either before or after the e-Go Congress, I want to take you up on your offer to pair on this. I'd really like to get webmock working for this.

**Comment by neagle** [comment]
Very excited to check this out--thanks again for taking this task on!

1. Checked out the branch
2. Ran `bundle install`
3. Ran `rake db:migrate`
4. Ran `rails server`
5. Navigated to `localhost:3000`
6. Got this error:

image

If I add the year to the URL, the page seems to load correctly... and now, mysteries of mysteries, when I go back to the root (`localhost:3000`), it seems to be working. 😓 

I also got an error when trying to sign up, possibly related to my local settings:

image

...I have `GMAIL_SMTP_USER` and `GMAIL_SMTP_PASSWORD` values in my `.env` file that I grabbed from Heroku; any advice on how the SMTP stuff should work locally?

Anyway -- sorry for the initial bumps, hopefully we can smooth 'em out.

**Comment by neagle** [comment]
Just tested out your branch and things worked great. One thought I had... when I signed up and checked my email, I immediately had two messages waiting for me:

image

The confirmation email, _and_ the welcome email. I think that, ideally, we'd like to shift the confirmation email to only go out when the account is actually confirmed, rather than when it's created. That way a user isn't confused about which email to look into for the account confirmation step. What do you think?

**Comment by jaredbeck** [comment]
> I think that, ideally, we'd like to shift the confirmation email to only go out when the account is actually confirmed, rather than when it's created.

Good catch. Done in b0494a660ec1d2badf6c80af1f8706500f3a60c9. Not tested in browser yet.

**Comment by neagle** [comment]
Closes #184. Thanks a lot for this feature, @jaredbeck!

**Pull request merged**